src/net/sourceforge/kolmafia/request/FightRequest.java

```diff
@@ -4387,6 +4386,28 @@ else if (responseText.contains("a horrible grinning clown head emerges")) {
     }
   }
 
+  private static void trackEnvironment(final KoLAdventure location) {
+    var environment = location != null ? location.getEnvironment() : "none";
+
+    var symbol =
+        switch (environment) {
+          case "outdoor" -> "o";
+          case "indoor" -> "i";
+          case "underground" -> "u";
+          case "underwater" -> "x";
+          default -> "?";
+        };
+
+    var environments = new StringBuilder(Preferences.getString("lastCombatEnvironments"));
+    environments.append(symbol);
+    while (environments.length() < 20) {
+      environments.insert(0, "x");
+    }
```

Comment:
Feel like `environments = "x".repeat(20 - environments.length()) + environments` or something like that is nicer.

Reply:
Agreed!

Reply:
I was even cheekier